### ISSUE TITLE
请整合个本地全文检索的功能

### ISSUE THREAD

**Issue by leeyaunlong** [altairwei/WizNotePlus#60]
请整合个本地全文检索的功能
比如solr+结巴. 虽然开销大,但是如果长期使用wiz,这带来的优势巨大.

**Comment by altairwei** [comment]
我查看了官方代码库，他们本打算使用 CLucene 来实现本地搜索的，只是后来没继续实现，搁置了。

**Comment by rengui** [comment]
现在没必要了吧，官方server的docker镜像放出来了，可以本地用。

ps, 它里面用的Elasticsearch做的全文搜索。

**Comment by altairwei** [comment]
> 现在没必要了吧，官方server的docker镜像放出来了，可以本地用。
> 
> ps, 它里面用的Elasticsearch做的全文搜索。

是的，Elasticsearch 非常优秀。本地全文搜索有个根本性的缺陷：文档数据集不完整，除非手动选择同步所有文档。

**Comment by V-ctor** [comment]
But aren't we  able to make full test search by SQLite engine? I understand that it won't be as perfect as Lucent based but it seems to be good enough for short time cases when we don't have connection to server. Doesn't it?

**Comment by altairwei** [comment]
> But aren't we able to make full test search by SQLite engine? I understand that it won't be as perfect as Lucent based but it seems to be good enough for short time cases when we don't have connection to server. Doesn't it?

Yes, full text search of SQLite is sufficient for local usage. Indeed, it has been planed to be implemented in v2.14
